File: api/Model/src/main/kotlin/sh/awtk/vothemis/viewmodel/Candidate.kt
package sh.awtk.vothemis.viewmodel

import sh.awtk.vothemis.dto.CandidateDto
import sh.awtk.vothemis.vo.CandidateDescription
import sh.awtk.vothemis.vo.CandidateId
import sh.awtk.vothemis.vo.NumOfVote
import sh.awtk.vothemis.vo.QuestionId

data class Candidate(
    val candidate_id: Long,
    val question_id: Long,
    val description: String,
    val num_of_vote: Long
)

fun Candidate.toDto(): CandidateDto = CandidateDto(
    id = CandidateId(candidate_id),
    questionId = QuestionId(question_id),
    description = CandidateDescription(description),
    numOfVote = NumOfVote(num_of_vote)
)

File: www/assets/scripts/model/Candidate.ts
export default class Candidate {
  get candidateId (): Number {
    return this._candidateId
  }

  set candidateId (value: Number) {
    this._candidateId = value
  }

  get questionId (): Number {
    return this._questionId
  }

  set questionId (value: Number) {
    this._questionId = value
  }

  get description (): String {
    return this._description
  }

  set description (value: String) {
    this._description = value
  }

  get numOfVote (): Number {
    return this._numOfVote
  }

  set numOfVote (value: Number) {
    this._numOfVote = value
  }

  get percentage (): Number {
    return this._percentage
  }

  set percentage (value: Number) {
    this._percentage = value
  }

  private _candidateId!:Number
  private _questionId!:Number
  private _description!:String
  private _numOfVote:Number = 0
  private _percentage:Number = 0

  constructor (candidateId:Number, questionId:Number, description:String, numOfVote:Number, percentage:Number) {
    this.candidateId = candidateId
    this.questionId = questionId
    this.description = description
    this.numOfVote = numOfVote
    this.percentage = percentage
  }
}
